examples/green_screen/main.cpp

```diff
@@ -0,0 +1,490 @@
+#include <chrono>
+#include <getopt.h>
+
+#include <k4a/k4a.hpp>
+#include <opencv2/calib3d.hpp>
+#include <opencv2/core.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+#include "transformation.h"
+#include "MultiDeviceCapturer.h"
+
+// Allowing at least 160 microseconds between depth cameras should ensure they do not interfere with one another.
+constexpr uint32_t MIN_TIME_BETWEEN_DEPTH_CAMERA_PICTURES_USEC = 160;
+
+// ideally, we could generalize this to many OpenCV types
+cv::Mat color_to_opencv(const k4a::image &im)
+{
+    // TODO this only handles mjpg
+    cv::Mat raw_data(1, im.get_size(), CV_8UC1, (void *)im.get_buffer());
+    cv::Mat decoded = cv::imdecode(raw_data, cv::IMREAD_COLOR);
+    return decoded;
+}
+
+cv::Mat depth_to_opencv(const k4a::image &im)
+{
+    return cv::Mat(im.get_height_pixels(),
+                   im.get_width_pixels(),
+                   CV_16U,
+                   (void *)im.get_buffer(),
+                   im.get_stride_bytes());
+}
+
+cv::Matx33f calibration_to_color_camera_matrix(const k4a::calibration &cal)
+{
+    const k4a_calibration_intrinsic_parameters_t::_param &i = cal.color_camera_calibration.intrinsics.parameters.param;
+    cv::Matx33f camera_matrix = cv::Matx33f::zeros();
+    camera_matrix(0, 0) = i.fx;
+    camera_matrix(1, 1) = i.fx;
+    camera_matrix(0, 2) = i.cx;
+    camera_matrix(1, 2) = i.cy;
+    camera_matrix(2, 2) = 1;
+    return camera_matrix;
+}
+
+Transformation get_depth_to_color_transformation_from_calibration(const k4a::calibration &cal)
+{
+    const k4a_calibration_extrinsics_t &ex = cal.extrinsics[K4A_CALIBRATION_TYPE_DEPTH][K4A_CALIBRATION_TYPE_COLOR];
+    Transformation tr;
+    for (int i = 0; i < 3; ++i)
+    {
+        for (int j = 0; j < 3; ++j)
+        {
+            tr.R(i, j) = ex.rotation[i * 3 + j];
+        }
+    }
+    tr.t = cv::Vec3d(ex.translation[0], ex.translation[1], ex.translation[2]);
+    return tr;
+}
+
+void set_calibration_depth_to_color_from_transformation(k4a::calibration &cal, const Transformation &tr)
+{
+    k4a_calibration_extrinsics_t &ex = cal.extrinsics[K4A_CALIBRATION_TYPE_DEPTH][K4A_CALIBRATION_TYPE_COLOR];
+    for (int i = 0; i < 3; ++i)
+    {
+        for (int j = 0; j < 3; ++j)
+        {
+            ex.rotation[i * 3 + j] = static_cast<float>(tr.R(i, j));
+        }
+    }
+    for (int i = 0; i < 3; ++i)
+    {
+        ex.translation[i] = static_cast<float>(tr.t[i]);
+    }
+}
+
+vector<float> calibration_to_color_camera_dist_coeffs(const k4a::calibration &cal)
+{
+    const k4a_calibration_intrinsic_parameters_t::_param &i = cal.color_camera_calibration.intrinsics.parameters.param;
+    return { i.k1, i.k2, i.p1, i.p2, i.k3, i.k4, i.k5, i.k6 };
+}
+
+// Takes the images by value so we can change them when shown.
+bool find_chessboard_corners_helper(cv::Mat main_color_image,
+                                    cv::Mat backup_color_image,
```

Comment:
>backup_color_image [](start = 44, length = 18)

I see that backup is used multiple times through this file should it be called "subordinate" or "secondary" instead?

Reply:
Good question. I use "master" and "subordinate" in the MultiDeviceCapturer file, but in this file it doesn't really matter that the camera is a subordinate in terms of device timing, but it matters a lot more that it's the camera that has depth that fills in the gaps for the main camera (it's not the one whose video feed we see).
I think "secondary" is hopefully more clear than "backup", though, so I switched it.